gym/envs/registration.py

```diff
@@ -47,12 +27,38 @@ def load(name: str) -> Type:
     return fn
 
 
+def parse_env_id(id: str) -> Tuple[Optional[str], str, Optional[int]]:
+    """Parse environment ID string format.
+
+    This format is true today, but it's *not* an official spec.
+    [username/](env-name)-v(version)    env-name is group 1, version is group 2
+
+    2016-10-31: We're experimentally expanding the environment ID format
+    to include an optional username.
+    """
+    pattern = re.compile(
```

Comment:
We should probably lift `pattern` outside of this function so it gets compiled once and cached between calls to `parse_env_id`. We do call this function quite a bit. Perhaps again naming it `env_id_re`?

Reply:
Sounds like a valid point!